stdlib/public/runtime/EnvironmentVariables.cpp

```diff
@@ -24,6 +25,12 @@ using namespace swift;
 
 namespace {
 
+// This is required to make the macro machinery work correctly; we can't
+// declare a VARIABLE(..., const char *, ...) because then the token-pasted
+// names won't work properly.  It *does* mean that if you want to use std::string
```

Comment:
Maybe name it something unlikely to collide, like `swift_const_char_ptr`?

Reply:
That would be fine, except that then the documentation you get from `SWIFT_HELP` will tell you the type is `const_char_ptr`. Like I said, someone went a bit OTT with macros.